buildSrc/src/main/java/org/elasticsearch/gradle/test/RestIntegTestTask.java

```diff
@@ -36,6 +36,8 @@
     private static final String TESTS_CLUSTER = "tests.cluster";
     private static final String TESTS_CLUSTER_NAME = "tests.clustername";
 
+    // TODO: refactor this so that work is not done in constructor and find usages and register them, not create them
+    // See: https://docs.gradle.org/current/userguide/task_configuration_avoidance.html
```

Comment:
FWIW, there's an open issue for this. Perhaps we can link it here.

https://github.com/elastic/elasticsearch/issues/47804

Reply:
done.